### PR TITLE
Cleaner eigencomponents for three-qubit gates

### PR DESCRIPTION
**Pull request by idk3**

### CODE REVIEW

**Comment by kevinsung** [comment]
I disagree with this change. If the rows fit on one line, then having the explicit matrix is clearer.

**Comment by Strilanc** [comment]
There's a block_diag method in cirq that might make this succinct while still retaining the structural details.

**Comment by idk3** [comment]
Hmm, that would be useful for the eigencomponents of https://github.com/quantumlib/OpenFermion-Cirq/pull/261 as well, I'll look into it.

File: openfermioncirq/gates/three_qubit_gates.py [hunk 1]
```diff
@@ -112,25 +112,16 @@ def half_turns(self) -> Union[cirq.Symbol, float]:
...
-                               [0, 0, 0, 0, 0, -0.5, 0.5, 0],
-                               [0, 0, 0, 0, 0, 0, 0, 0]]))
-        ]
+        minus_half_component = cirq.linalg.block_diag(
+            numpy.diag([0] * 5), numpy.array([[0.5, 0.5], [0.5, 0.5]]),
+            numpy.diag([0]))
+        plus_half_component = cirq.linalg.block_diag(
+            numpy.diag([0] * 5), numpy.array([[0.5, -0.5], [-0.5, 0.5]]),
+            numpy.diag([0]))
+
```
Comment (kevinsung) [inline openfermioncirq/gates/three_qubit_gates.py]: Expand the zeros out and put each block on its own line.
```
        minus_half_component = cirq.linalg.block_diag(
            numpy.diag([0, 0, 0, 0, 0] ),
            numpy.array([[0.5, 0.5], [0.5, 0.5]]),
            numpy.diag([0]))

Comment (idk3) [reply openfermioncirq/gates/three_qubit_gates.py]: Done - what's your tolerable limit for number of zeros before you'd convert to `numpy.diag([0] * n)`? I find these get hard to read pretty quickly (~5 is fine, but close to my limit).

**Pull request merged**